lib/gemstash/http_client.rb

```diff
@@ -59,8 +51,28 @@ def get(path)
       end
     end
 
+    def head(path)
+      response = request(:head, path)
+      if block_given?
+        yield response.headers
+      else
+        response.headers
+      end
+    end
+
   private
 
+    def request(method, path)
+      response = with_retries do
+        @client.public_send(method, path) do |request|
+          request.headers["User-Agent"] = @user_agent
+          request.options.open_timeout = 2
+        end
+      end
+      raise Gemstash::WebError.new(response.body, response.status) unless response.success?
```

Comment:
With a head request, does the body come back empty or null? If so, should we provide some sensible default message when that is the case? Keep in mind that the message is likely to be displayed to the user though, so if it is somehow useful to them, that would be extra helpful